cypress/integration/ui/create_dashboard/index.js

```diff
@@ -0,0 +1,111 @@
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps'
+
+const UID_LENGTH = 11
+
+const dashboardTitle = new Date().toString()
+
+const ROUTE_EDIT = 'edit'
+const ROUTE_NEW = 'new'
+const ROUTE_PRINTLAYOUT = 'printlayout'
+const ROUTE_PRINTOIPP = 'printoipp'
+const nonViewRoutes = [
+    ROUTE_NEW,
+    ROUTE_EDIT,
+    ROUTE_PRINTLAYOUT,
+    ROUTE_PRINTOIPP,
+]
+
+const getRouteFromHash = hash => {
+    const lastSlashIdx = hash.lastIndexOf('/')
+    return hash.slice(lastSlashIdx + 1)
+}
+
+const toggleShowMoreButton = () => {
+    cy.get('[data-test="dhis2-dashboard-showmore-button"]').click()
+}
+
+beforeEach(() => {
+    cy.visit('/')
+})
+
+Given('I choose to create new dashboard', () => {
+    cy.get('[data-test="dhis2-dashboard-link-new-dashboard"]').click()
+})
+
+When('dashboard title is added', () => {
+    cy.get('[data-test="dhis2-dashboard-dashboard-title-input"]').type(
+        dashboardTitle
+    )
+})
+
+When('dashboard items are added', () => {
+    cy.get('[data-test="dhis2-dashboard-item-search"]').click()
+    cy.get(
+        '[data-test="dhis2-dashboard-menu-item-ANC: 1 and 3 coverage Yearly"]'
+    ).click()
+    // close the item selector
+    cy.get('body').trigger('keydown', { key: 'Escape' })
```

Comment:
Extract this to a separate step e.g. "and the item selector is closed" that also asserts that the item selector is no longer visible?

Reply:
Done